tensorflow/python/ops/variable_scope.py

```diff
@@ -662,8 +662,13 @@ def _get_single_variable(self,
         init_val = initializer
         variable_dtype = None
       else:
-        init_val = lambda: initializer(
-            shape.as_list(), dtype=dtype, partition_info=partition_info)
+        if partition_info is not None:
```

Comment:
You can fix this either by fixing `sigma_init`'s signature to use `partition_info=None`, or by explicitly checking here how many arguments the function has and passing partition_info if it has three. This is a mess though since the inspect module has changed a bit between python2 and python3, and you have to make sure that you can deal with all callables: objects that define __call__, functions, and lambdas.